vendor/github.com/openshift/library-go/pkg/operator/encryption/controllers/migrators/errors.go

```diff
@@ -21,73 +21,27 @@ import (
 
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/utils/pointer"
 )
 
-// ErrRetriable is a wrapper for an error that a migrator may use to indicate the
-// specific error can be retried.
-type ErrRetriable struct {
-	error
-}
-
-func (ErrRetriable) Temporary() bool { return true }
-
-// ErrNotRetriable is a wrapper for an error that a migrator may use to indicate the
-// specific error cannot be retried.
-type ErrNotRetriable struct {
-	error
-}
-
-func (ErrNotRetriable) Temporary() bool { return false }
-
-// TemporaryError is a wrapper interface that is used to determine if an error can be retried.
-type TemporaryError interface {
-	error
-	// Temporary should return true if this is a temporary error
-	Temporary() bool
-}
-
 // isConnectionRefusedError checks if the error string include "connection refused"
 // TODO: find a "go-way" to detect this error, probably using *os.SyscallError
 func isConnectionRefusedError(err error) bool {
 	return strings.Contains(err.Error(), "connection refused")
 }
 
-// interpret adds retry information to the provided error. And it might change
-// the error to nil.
-func interpret(err error) error {
+// canRetry returns false if the provided error indicates a retry is
+// impossible. It returns true if the error is possibly temporary. It returns
+// nil for all other error where it is unclear.
+func canRetry(err error) *bool {
```

Comment:
why do we want to return a pointer?

Reply:
because we have three values: retryable, not retryable, don't know